src/applications/personalization/profile360/containers/Vet360TransactionReporter.jsx

```diff
@@ -26,39 +28,43 @@ class Vet360TransactionReporter extends React.Component {
     if (newMessageVisible) scrollToTop();
   }
 
+  clearAllSuccessfulTransactions = () => {
+    this.props.successfulTransactions.forEach(this.props.clearTransaction);
+  }
+
+  clearAllErroredTransactions = () => {
+    this.props.erroredTransactions.forEach(this.props.clearTransaction);
+  }
+
   render() {
     const {
-      successfulTransactions,
-      erroredTransactions
+      mostRecentSuccessfulTransaction,
+      mostRecentErroredTransaction
     } = this.props;
 
     return (
       <div className="vet360-transaction-reporter">
-        {successfulTransactions.map((transaction) => {
-          return (
-            <AlertBox
-              key={transaction.data.attributes.transactionId}
-              isVisible
-              status="success"
-              onCloseAlert={this.props.clearTransaction.bind(null, transaction)}
-              content={<h4>We saved your updated information.</h4>}/>
-          );
-        })}
-        {erroredTransactions.map((transaction) => {
-          return (
-            <Vet360TransactionErrorBanner
-              key={transaction.data.attributes.transactionId}
-              transaction={transaction}
-              clearTransaction={this.props.clearTransaction.bind(null, transaction)}/>
-          );
-        })}
+        {mostRecentSuccessfulTransaction && (
+          <AlertBox
+            isVisible
+            status="success"
+            onCloseAlert={this.clearAllSuccessfulTransactions}
+            content={<h4>We saved your updated information.</h4>}/>
+        )}
+        {mostRecentErroredTransaction && (
+          <Vet360TransactionErrorBanner
+            transaction={mostRecentErroredTransaction}
+            clearTransaction={this.clearAllErroredTransactions}/>
+        )}
       </div>
     );
   }
 }
 
 const mapStateToProps = (state) => {
   return {
+    mostRecentSuccessfulTransaction: selectMostRecentSuccessfulTransaction(state),
+    mostRecentErroredTransaction: selectMostRecentErroredTransaction(state),
     successfulTransactions: selectVet360SuccessfulTransactions(state),
```

Comment:
Added new selectors for grabbing just the most recent successful/errored transaction. However, we still need to get the corresponding arrays, bc we need to clear all of them once the user clicks the close button.